pep-0548.rst

```diff
@@ -0,0 +1,338 @@
+==============================================================
+
+* cross-platform applications should use appropriate mechanisms (such as Python
+  virtual environments or system level application containers) to avoid
+  depending on the default version of Python installed on any particular
+  deployment platform.
+
+
+Recommendations for Platform Providers
+======================================
+
+* Unix-like software distributions (including systems like Mac OS X and
+  Cygwin) should install the ``python2`` command into the default path
+  whenever a version of the Python 2 interpreter is installed, and the same
+  for ``python3`` and the Python 3 interpreter.
+* When  invoked, ``python2`` should run some version of the Python 2
+  interpreter, and ``python3`` should run some version of the Python 3
+  interpreter.
+* The more general ``python`` command should be handled using one of the
+  following techniques:
+
+  * if present, always refers to the same target as ``python2``
+    (e.g. Debian 7/8/9, RHEL/CentOS 5/6/7, Ubuntu LTS 14.04 & 16.04)
+  * if present, always refers to the same target as ``python3``
+    (e.g. Arch, OpenMandriva)
+  * if present, refers to the same target as ``python3`` by default, can be
+    switched by a system administrator to refer to the same target as
+    ``python2`` (e.g. Gentoo)
+  * absent by default (even when ``python2`` and ``python3`` are available),
+    but system administrators can explicitly opt-in to having it refer to either
+    ``python2`` (for full compatibility with Python-2-only distributions) or
+    ``python3`` (for compatibility with cross-version shebang lines without
+    installing a second Python runtime environment) (e.g. proposal currently
+    in development for Fedora 27+)
+
+Additional recommendations when providing a ``python`` command:
+
+* The Python 2.x ``idle``, ``pydoc``, and ``python-config`` commands should
+  be made available as ``idle2``, ``pydoc2``, and ``python2-config``,
+  with the original unqualified commands invoking either ``python2`` or
+  ``python3`` in accordance with the target version for ``python``.
+* Regardless of whether it refers to ``python2`` or ``python3``, if a ``python``
+  link is provided the target Python stack should include the``six``
+  compatibility library by default, as having this library available
+  significantly increases the common subset of Python 2.7 and 3.6+.
+* If a ``python`` link targets ``python3``, the target version should be at
+  least 3.6+ in order to provide access to:
+
+  * Python 2 style explicit Unicode literals (restored in 3.3)
+  * convenience aliases for binary codecs (restored in 3.4)
+  * printf-style formatting on bytes objects: ``b"%s: %d" % (field_name, value)``
+    (restored in 3.5)
+  * improved error messages for Python 2 style print statements (initially
+    improved in 3.4.2, subsequently enhanced further with suggested calls to
+    the print builtin in 3.6.3)
```

Comment:
Is this important enough to floor on Python 3.6?  Maybe, especially since it is the current stable release.  3.5 might be EOL'd by the time this PEP takes effect (I haven't done the math ;).